src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxIndexFacetedReadOperation.cs

```diff
@@ -0,0 +1,428 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Threading;
+using Corax;
+using Nest;
+using Raven.Client.Documents.Indexes;
+using Raven.Client.Documents.Queries.Facets;
+using Raven.Server.Documents.Indexes.Persistence.Lucene;
+using Raven.Server.Documents.Indexes.Static.Spatial;
+using Raven.Server.Documents.Queries;
+using Raven.Server.Documents.Queries.Facets;
+using Raven.Server.Documents.Queries.Timings;
+using Raven.Server.ServerWide.Context;
+using Sparrow;
+using Sparrow.Logging;
+using Sparrow.Server;
+using Voron;
+using Voron.Data.BTrees;
+using Voron.Impl;
+using IndexEntryReader = Corax.IndexEntryReader;
+using RangeType = Raven.Client.Documents.Indexes.RangeType;
+using Token = Corax.Pipeline.Token;
+
+namespace Raven.Server.Documents.Indexes.Persistence.Corax;
+
+public class CoraxIndexFacetedReadOperation : IndexFacetReadOperationBase
+{
+    private readonly IndexFieldsMapping _fieldMappings;
+    private readonly IndexSearcher _indexSearcher;
+    private readonly ByteStringContext _allocator;
+    private readonly Dictionary<string, Slice> _fieldNameCache;
+
+    public CoraxIndexFacetedReadOperation(Index index, Logger logger, Transaction readTransaction, QueryBuilderFactories queryBuilderFactories,
+        IndexFieldsMapping fieldsMapping) : base(index, queryBuilderFactories, logger)
+    {
+        _fieldMappings = fieldsMapping;
+        _allocator = readTransaction.Allocator;
+        _fieldMappings = fieldsMapping;
+        _indexSearcher = new IndexSearcher(readTransaction, _fieldMappings);
+        _fieldNameCache = new();
+    }
+
+    public override List<FacetResult> FacetedQuery(FacetQuery facetQuery, QueryTimingsScope queryTimings, DocumentsOperationContext context,
+        Func<string, SpatialField> getSpatialField, CancellationToken token)
+    {
+        var results = FacetedQueryParser.Parse(context, facetQuery, SearchEngineType.Corax);
```

Comment:
Misleading name, those are the facets, no?

Reply:
Well, yes. Exactly the same name as in `Lucene` reader. I've used it to make it simpler for a reader.